include/boost/wave/util/flex_string.hpp

```diff
@@ -1464,10 +1464,10 @@ class flex_string : private Storage
     typedef typename A::size_type size_type;
     typedef typename A::difference_type difference_type;
 
-    typedef typename A::reference reference;
-    typedef typename A::const_reference const_reference;
-    typedef typename A::pointer pointer;
-    typedef typename A::const_pointer const_pointer;
+    typedef typename A::value_type& reference;
+    typedef typename A::value_type const& const_reference;
+    typedef typename A::value_type* pointer;
+    typedef typename A::value_type const* const_pointer;
```

Comment:
Wouldn't it be better to change this to using `std::allocator_traits` (if available)?

Reply:
Thanks for your remarks. I update it.